src/GuessControl.js

```diff
@@ -1,47 +1,41 @@
-import React, { Component } from "react";
+import React, { useEffect, useState } from "react";
 import Button from "./Button";
 
-class GuessControl extends Component {
-  constructor(props) {
-    super(props);
+function GuessControl({ onGuess }) {
+	const [currentGuess, setCurrentGuess] = useState("");
 
-    this.state = {
-      currentGuess: "",
-    };
+	const handleInputChange = (event) => {
+		setCurrentGuess(event.target.value);
+	};
 
-    /**
-     * These lines are required to make the methods/functions declared on this
-     *  class have the correct `this` object when they run.
-     */
-    this.handleInputChange = this.handleInputChange.bind(this);
-    this.onSubmitGuess = this.onSubmitGuess.bind(this);
-  }
+	const onSubmitGuess = () => {
+		onGuess(Number(currentGuess));
+		setCurrentGuess("");
+	};
+	// Allow user to use 'Enter' key to submit guess
+	useEffect(() => {
+		const listener = (event) => {
+			if (event.code === "Enter" || event.code === "NumpadEnter") {
+				onGuess(Number(currentGuess));
+				setCurrentGuess("");
+			}
+		};
+		document.addEventListener("keydown", listener);
+		return () => {
+			document.removeEventListener("keydown", listener);
+		};
+	}, [onGuess, currentGuess]);
```

Comment:
Pretty interesting approach. This use effect is a bit heavy-handed but does what it needs to do. Since this is an exercise, and your experimenting, this is okay.

This is not a part of the conversion exercise, but the more appropriate thing would have been to turn the contents of this component into a controlled form. `onSubmit` on the form listens to both an input with type `submit` and the enter button.

Reply:
Oh I see! Thanks!